From 7e91f24c73256af59ac9061c41b73a184c4690aa Mon Sep 17 00:00:00 2001
From: Stefan Eissing <stefan@eissing.org>
Date: Mon, 3 Nov 2025 15:07:57 +0100
Subject: [PATCH] cw-out: fix EAGAIN handling on pause

The interim CURLE_AGAIN result was not always converted to a
CURLE_OK and then caused write callers to report a failure.

Fixes #19334
Reported-by: pennae on github
Closes #19338
---
 lib/cw-out.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/cw-out.c b/lib/cw-out.c
index 1f4031649267..9c0a36e7e5e2 100644
--- a/lib/cw-out.c
+++ b/lib/cw-out.c
@@ -302,6 +302,7 @@ static CURLcode cw_out_buf_flush(struct cw_out_ctx *ctx,
                               &consumed);
     if(result && (result != CURLE_AGAIN))
       return result;
+    result = CURLE_OK;
 
     if(consumed) {
       if(consumed == curlx_dyn_len(&cwbuf->b)) {
